### PR TITLE
readRenderTargetPixels: add parameter to select active cube face from WebGLRenderTargetCube

### LINKED ISSUE #16331: readRenderTargetPixels not working with WebGLRenderTargetCube
##### Description of the problem

I've experienced that `readRenderTargetPixels` is not working with `WebGLRenderTargetCube`, e.g.

```js
renderer.setRenderTarget(RT, i);
renderer.readRenderTargetPixels(RT, 0, 0, RT.width, RT.height, pixelBuffer);
```

I guess the `setRenderTarget` call isn't having an effect. There is a getter in `readRenderTargetPixels` that returns an array of six `WebGLFramebuffer`.

```js
	this.readRenderTargetPixels = function ( renderTarget, x, y, width, height, buffer ) {

		if ( ! ( renderTarget && renderTarget.isWebGLRenderTarget ) ) {

			console.error( 'THREE.WebGLRenderer.readRenderTargetPixels: renderTarget is not THREE.WebGLRenderTarget.' );
			return;

		}

		var framebuffer = properties.get( renderTarget ).__webglFramebuffer;

		if ( framebuffer ) {

			var restore = false;

			if ( framebuffer !== _currentFramebuffer ) {

				_gl.bindFramebuffer( 36160, framebuffer );

				restore = true;

			}
```

What i'm trying to achieve is for a given cube envmap to extract the cube levels of detail coming from `PMREMGenerator` given in the HDR envmap examples. Are there any workarounds for this?

##### Three.js version

- [ ] r102

### PR DESCRIPTION
**Pull request by throni3git**
This solves the issue that you can't readRenderTargetPixels with a WebGLRenderTargetCube.
Fixes https://github.com/mrdoob/three.js/issues/16331

### CODE REVIEW

File: src/renderers/WebGLRenderer.js [hunk 1]
```diff
@@ -2512,6 +2512,12 @@ function WebGLRenderer( parameters ) {
 
 		var framebuffer = properties.get( renderTarget ).__webglFramebuffer;
 
+		if ( activeCubeFaceIndex != undefined ) {
```
Comment (mrdoob) [inline src/renderers/WebGLRenderer.js]: Should we check if `renderTarget.isWebGLRenderTargetCube`?

Comment (throni3git) [reply src/renderers/WebGLRenderer.js]:
```suggestion
		if ( renderTarget.isWebGLRenderTargetCube && activeCubeFaceIndex != undefined ) {
```
Yes. In my projects i often use `instanceof` to check an object being of a certain class. I forgot that there are these flags :)

File: src/renderers/WebGLRenderer.js [hunk 2]
```diff
@@ -2512,6 +2512,12 @@ function WebGLRenderer( parameters ) {
 
 		var framebuffer = properties.get( renderTarget ).__webglFramebuffer;
 
+		if ( renderTarget.isWebGLRenderTargetCube && activeCubeFaceIndex != undefined ) {
```
Comment (Mugen87) [inline src/renderers/WebGLRenderer.js]: Minor code style issue:
```js
if ( renderTarget.isWebGLRenderTargetCube && activeCubeFaceIndex !== undefined ) {
```

Comment (throni3git) [reply src/renderers/WebGLRenderer.js]: Yeah sure, thanks.

Comment (throni3git) [reply src/renderers/WebGLRenderer.js]:
```suggestion
		if ( renderTarget.isWebGLRenderTargetCube && activeCubeFaceIndex !== undefined ) {
```

Comment (Mugen87) [reply src/renderers/WebGLRenderer.js]: Sry, I don't have the permission to commit the suggestion.

Comment (throni3git) [reply src/renderers/WebGLRenderer.js]: Well, this is my first pull request^^
I committed the change...

**Review by Mugen87 — APPROVED** [review]

**Comment by throni3git** [comment]
So, can i close this pull request now?

**Comment by Mugen87** [comment]
> So, can i close this pull request now?

Closing would mean the PR is not merged and your changes to not land in the library. @mrdoob will perform the merge after a final positive review.

**Comment by throni3git** [comment]
Thanks for stating that :)

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**